x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/sagemaker/SageMakerService.java

```diff
@@ -74,6 +79,30 @@ public SageMakerService(
                 .setConfigurations(configurationMap.get())
                 .build()
         );
+        this.clusterService = Objects.requireNonNull(context.clusterService());
+    }
+
+    // for testing
+    public SageMakerService(
+        SageMakerModelBuilder modelBuilder,
+        SageMakerClient client,
+        SageMakerSchemas schemas,
+        ThreadPool threadPool,
+        CheckedSupplier<Map<String, SettingsConfiguration>, RuntimeException> configurationMap,
+        ClusterService clusterService
+    ) {
+        this.modelBuilder = modelBuilder;
```

Comment:
Could you call this(.... context.clusterService()) in the other constructor instead of duplicating initialization logic?